.github/workflows/publish.yml

```diff
@@ -0,0 +1,39 @@
+name: Publish to npm
+
+on:
+  workflow_run:
+    workflows: ["Main"]
+    types:
+      - completed
+    branches: [ "main" ]
+
+jobs:
+  publish:
+    if: ${{ github.event.workflow_run.conclusion == 'success' }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version: '20'
+          registry-url: 'https://registry.npmjs.org/'
+      - uses: oven-sh/setup-bun@v1
+      - run: bun install
+      - name : update changelog
+        id: update-changelog
+        run: bun run update-changelog
```

Comment:
run update changelog script and use exit code 0 or 1 to know how to proceed next